pkg/resource/provider/component_provider.go

```diff
@@ -32,6 +32,46 @@ type componentProvider struct {
 	version   string
 	schema    []byte
 	construct provider.ConstructFunc
+	call      provider.CallFunc
+}
+
+type ProviderOption func(*componentProvider)
+
+func WithConstructFunc(f provider.ConstructFunc) ProviderOption {
```

Comment:
If we stick with functional options, should these be simply `WithConstruct` and `WithCall` without the `Func` suffix?

Reply:
I'd say no, since `WithConstruct` in this case would be used for an object that implemented the interface (similar to ServeFunc etc in the sodlib).